src/devices/cpu/m6502/gew12.cpp

```diff
@@ -0,0 +1,229 @@
+// license:BSD-3-Clause
+// copyright-holders:Devin Acker
+/***************************************************************************
+
+    gew7.cpp
+
+    Yamaha GEW2 (65c02-based)
+
+***************************************************************************/
+
+#include "emu.h"
+#include "gew12.h"
+
+#include "m6502mcu.ipp"
+
+DEFINE_DEVICE_TYPE(GEW12, gew12_device, "gew12", "Yamaha YMW728-F (GEW12)")
+
+gew12_device::gew12_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: m6502_mcu_device_base<m65c02_device>(mconfig, GEW12, tag, owner, clock)
+	, device_mixer_interface(mconfig, *this, 2)
+	, m_in_cb(*this), m_out_cb(*this)
+	, m_rom(*this, DEVICE_SELF)
+	, m_bank(*this, "bank%u", 0U)
+{
+	program_config.m_internal_map = address_map_constructor(FUNC(gew12_device::internal_map), this);
+}
+
+
+void gew12_device::device_add_mconfig(machine_config &config)
+{
+	// TODO: PCM, UART
+}
+
+void gew12_device::device_start()
+{
+	m65c02_device::device_start();
+
+	m_in_cb.resolve_all_safe(0);
+	m_out_cb.resolve_all_safe();
```

Comment:
Might it be better to use `.resolve_all_safe(0xff)` for the inputs?   Unused inputs are more likely to have weak pull-ups if they don’t have bus-friendly logic that weakly holds the last level.